From 0034219eec60cd191a4ff7d6c9fc1f21c73ee704 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Tue, 21 Jan 2014 10:47:59 +0800
Subject: [PATCH 437/505] ENGR00295894 PXP_TEST: Initialize the pxp_mem_desc
 data before passing to driver

The allocated pxp_mem_desc data should be intialized before passing
to pxp driver, since the driver will use the mtype field to record
the requested physical memory type. If don't intialized, the driver
may get a random value from mtype.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 test/pxp_lib_test/pxp_test.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/test/pxp_lib_test/pxp_test.c b/test/pxp_lib_test/pxp_test.c
index 6ac13b9..1fd5983 100644
--- a/test/pxp_lib_test/pxp_test.c
+++ b/test/pxp_lib_test/pxp_test.c
@@ -1,7 +1,7 @@
 /*
  * pxp_test - test application for the PxP DMA ENGINE lib
  *
- * Copyright (C) 2010-2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2014 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -255,9 +255,11 @@ int pxp_test(void *arg)
 		dbg(DBG_ERR, "pxp request channel err\n");
 		goto err0;
 	}
-	dbg(DBG_INFO, "requested chan_id %d\n", pxp_chan.chan_id);
+	dbg(DBG_INFO, "requested chan handle %d\n", pxp_chan.handle);
 
 	/* Prepare the channel parameters */
+	memset(&mem, 0, sizeof(struct pxp_mem_desc));
+	memset(&mem_o, 0, sizeof(struct pxp_mem_desc));
 	mem.size = WIDTH * HEIGHT * 2;
 	mem_o.size = WIDTH * HEIGHT;
 
-- 
1.8.0

